### PR TITLE
Carousel - Fix issue where next button shows when there is no more items

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

* Noticed an issue where the Next button would show when there are no more items in the carousel
* Added test for ensuring only `<Carousel.Item />` children components will render in the carousel

## Test Steps

1. Checkout branch - `git checkout carousel-updates`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the button storybook documentation

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed relevant documentation has been updated/added.
- [x] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

### CODE REVIEW

File: .github/workflows/pull-requests-test-and-lint.yml [hunk 1]
```diff
@@ -41,6 +41,9 @@ jobs:
       - name: Check linting on all files
         run: npm run lint
 
+      - name: Stylelint Owner Fix
+        run: sudo chown -R root:root $GITHUB_WORKSPACE
```
Comment (JackHowa) [inline .github/workflows/pull-requests-test-and-lint.yml:47]: woah what's happening here? Is there docs showing that this needs to be done?

File: src/components/carousel/index.jsx [hunk 2]
```diff
@@ -128,7 +130,7 @@ const Carousel = ({
 
 			<div className="c-carousel__actions">
 				{slide !== slidesToShow ? resolvedPreviousButton : null}
-				{slide !== carouselItems.length ? resolvedNextButton : null}
+				{slide !== carouselItems.length && carouselItems.length > 1 ? resolvedNextButton : null}
```
Comment (JackHowa) [inline src/components/carousel/index.jsx:133]: makes sense

File: src/components/carousel/index.test.jsx [hunk 3]
```diff
@@ -55,7 +71,20 @@ describe("Carousel", () => {
...
+		render(
+			<Carousel id="carousel-2" label="Carousel Label" slidesToShow={1}>
+				<Carousel.Item label="Slide 1 of 2">
+					<div />
+				</Carousel.Item>
+			</Carousel>
+		);
+
+		expect(screen.getByRole("region", { name: "Carousel Label" })).not.toBeNull();
+		expect(screen.queryAllByText("Next")).toHaveLength(0);
```
Comment (JackHowa) [inline src/components/carousel/index.test.jsx:87]: nice work

**Review by JackHowa — APPROVED** [review]
Good work

File: .github/workflows/pull-requests-test-and-lint.yml (hunk 1 above)
Comment (matthewroach) [reply .github/workflows/pull-requests-test-and-lint.yml:47]: Based on this comment - https://github.com/reviewdog/reviewdog/issues/1158#issuecomment-1097649732

Comment (JackHowa) [reply .github/workflows/pull-requests-test-and-lint.yml:47]: gotcha cool

**Comment by JackHowa** [comment]
extra comment looks great @matthewroach https://github.com/WPMedia/arc-themes-components/pull/65/commits/b234f9da19819a6a1314fa99130cf86a6021693d

**Pull request merged**